catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/model/FilterNode.java

```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.forms.model;
+
+import java.util.List;
+import java.util.Map;
+
+public interface FilterNode {
+
+  boolean isLeaf();
```

Comment:
Curious on whether we should throw javadoc on this interface.  Since it's just getters i think it is self explanatory

Reply:
Should we annotate some of these with @nullable?  It looks like that is possible looking at the FilterNodeImpl

Reply:
At some point I need to refactor this. We only need one implementation at this point. And this interface will go away. But I think this PR has seen too much change. I'll wait until this gets merged. For now, adding some javadoc to clarify.

Reply:
Found some implementation issues as well. Fixed.